tissuumaps/static/js/utils/filterUtils.js

```diff
@@ -275,13 +275,64 @@
                     }
                 }
             }
+        },
+        "Min":{
+            params:{
+                type:"range",
+                min:0,
+                max:255,
+                step:1,
+                value:0
+            },
+            filterFunction: function (value) {
+                if (value == 0) {  return function (context, callback) {callback();}}
+                return RESCALE(value,filterUtils.getFilterParams('Max').value);
+            }
+        },
+        "Max":{
+            params:{
+                type:"range",
+                min:0,
+                max:255,
+                step:1,
+                value:255
+            },
+            filterFunction: function (value) {
+                if (value == 255) {  return function (context, callback) {callback();}}
+                return RESCALE(filterUtils.getFilterParams('Min').value,value);
+            }
         }
     },
     _filterItems:{},
     _lastFilters:{},
     _compositeMode:"source-over"
 }
-
+RESCALE = function(vmin,vmax) {
```

Comment:
Did you try a comparison between using the canvas directly like you did, and using the `process` function from CalmanJS (see Pixel-wise filters here: http://camanjs.com/guides/#Extending or the `SplitChannel` function here: https://github.com/TissUUmaps/TissUUmaps/blob/master/tissuumaps/static/js/utils/filterUtils.js#L285)? I like the precomputedContrast used here, not sure we could do the same with the `process` function.

Reply:
Yes,that was actually the first thing that I tried.  I had the same bug that I described in https://github.com/TissUUmaps/TissUUmaps/issues/26#issue-1810672076

Reply:
Ok I see, I need to dig a bit more on the filter bugs. I will merge this as is, it works very well!

Reply:
Thanks!